app/assets/javascripts/discourse/app/controllers/delete-topic-confirm.js

```diff
@@ -0,0 +1,35 @@
+import I18n from "I18n";
+import discourseComputed from "discourse-common/utils/decorators";
+import Controller, { inject } from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+
+// Modal that displays confirmation text when user deletes a topic
+// The modal will display only if the topic exceeds a certain amount of views
+export default Controller.extend(ModalFunctionality, {
+  topicController: inject("topic"),
+  deleting: false,
+
+  @discourseComputed("deleting")
+  buttonDisabled(deleting) {
+    return !!deleting;
+  },
+
+  actions: {
+    deleteTopic() {
+      this.set("deleting", true);
+
+      this.topicController.model
+        .destroy(this.currentUser)
+        .then(() => {
+          this.send("closeModal");
+          this.set("deleting", false);
```

Comment:
I think it would be a bit better if you reset this property in `onShow()`. We try to reset the controllers before reusing them to get rid of any undesired state.

Check how it is used in other modal controllers.